readthedocs/oauth/services/github.py

```diff
@@ -311,6 +312,83 @@ def update_webhook(self, project, integration):
             )
             return (False, resp)
 
+    def send_build_status(self, project, identifier, state, build_url):
+        """
+        Create GitHub commit status for project.
+
+        :param project: project to set up commit status for
+        :type project: Project
+        :param identifier: commit sha of the version
+        :type identifier: str
+        :param state: build state failure, pending, or success.
+        :type state: str
+        :param build_url: build url
+        :type build_url: str
+        :returns: boolean based on commit status creation was successful or not.
+        :rtype: Bool
+        """
+        session = self.get_session()
+        owner, repo = build_utils.get_github_username_repo(url=project.repo)
+
+        # select the correct state and description.
+        github_build_state = SELECT_BUILD_STATUS[state]['github']
+        description = SELECT_BUILD_STATUS[state]['description']
+
+        data = {
+            "state": github_build_state,
+            "target_url": build_url,
+            "description": description,
+            "context": "continuous-documentation/read-the-docs"
+        }
+
+        resp = None
+        try:
+            resp = session.post(
+                (
+                    'https://api.github.com/repos/{owner}/{repo}/statuses/{sha}'
+                    .format(owner=owner, repo=repo, sha=identifier)
+                ),
+                data=json.dumps(data),
+                headers={'content-type': 'application/json'},
+            )
+            if resp.status_code == 201:
+                log.info(
+                    'GitHub commit status for project: %s',
+                    project,
+                )
+                return True
+
+            if resp.status_code in [401, 403, 404]:
+                log.info(
+                    'GitHub project does not exist or user does not have '
+                    'permissions: project=%s',
+                    project,
+                )
+                return False
+
+        # Catch exceptions with request or deserializing JSON
+        except (RequestException, ValueError):
+            log.exception(
+                'GitHub commit status creation failed for project: %s',
+                project,
+            )
+        else:
```

Comment:
Having an `else` for an error case feels weird. Is there a reason we're doing it this way? Why not just have it as part of the `try` block after the `return`'s?

Reply:
We were doing this in some other methods in the class. So, I did it to be consistent. But my own opinion is we don't really need it like this :)